streampipes-data-explorer/src/main/java/org/apache/streampipes/dataexplorer/query/DataExplorerQueryExecutor.java

```diff
@@ -53,23 +54,23 @@ public DataExplorerQueryExecutor(int maximumAmountOfEvents) {
   public SpQueryResult executeQuery(SelectQueryParams params,
                                     boolean ignoreMissingValues) throws RuntimeException {
 
+    X query = makeSelectQuery(params);
+    var result = executeQuery(query, ignoreMissingValues);
     if (this.maximumAmountOfEvents != -1) {
-      X countQuery = makeCountQuery(params);
-      W countQueryResult = executeQuery(countQuery);
       var limit = params.getLimit();
-      var amountOfResults = getAmountOfResults(countQueryResult);
-      Double amountOfQueryResults = limit == Integer.MIN_VALUE ? amountOfResults : Math.min(amountOfResults, limit);
-
+      var amountOfResults = result.getAllDataSeries().stream().mapToInt(DataSeries::getTotal).sum();
```

Comment:
Is it possible to extract some of the functionality of this method into private methods to make it easier to understand?